### DISCUSSION TITLE
How to migrate my existing data from mysql database to pocketbase ?

### DISCUSSION THREAD

**Discussion by samimhamza-pgl** [pocketbase/pocketbase#2088] [Q&A] [answered]
I have a builtin system that was created in laravel and mysql. now I want to migrate that system to pocketbase. but now I can't find any solution for data migration

**Comment by ganigeorgiev** [comment, marked as answer]
At the moment there is no built-in solution for this but eventually data import from json/csv will be implemented as part of #48 (_for now is on hold until the other [major and blocker tasks](https://github.com/orgs/pocketbase/projects/2) are completed_).

But in general I don't think a fully automated solution can be implemented because there are some PocketBase specifics (eg. how file fields are handled, how multi-valued fields are stored, how constraints and data validations are managed, etc.) which other dbs don't have or implement differently and a better approach might be to first manually design your database based on the available field options (eg. from the Admin UI), so that you can make use of all PocketBase features.

Once the collections are created, you could prepare the import script with data that is adapted for your new  structure.

You can execute the crafted import script directly against the `pb_data/data.db` SQLite db file (eg. using the [`sqlite3` cli tool](https://sqlite.org/cli.html#importing_files_as_csv_or_other_formats), dbeaver, etc.).
Or alternatively, if the data to import is in JSON, you can insert them with the web APIs (this allows finer control but might be slow depending on how big is your dataset).

Here is an example using the JS SDK from similar question in [#2014](https://github.com/pocketbase/pocketbase/discussions/2014):
```js
import PocketBase from "pocketbase";

const pb = new PocketBase("http://127.0.0.1:8090");

const items = [
    {name: "test1"},
    {name: "test2"},
    {name: "test3"},
    // ...
];


// if you want to preserve the insertion order, await each request
for (let item of items) {
    await pb.collection("example").create(item);
}

// ---

// or if you don't care about the insertion order, you can insert them concurrently
const promises = [];
for (let item of items) {
    promises.push(pb.collection("example").create(item, {
        "$autoCancel": false,
    }));
}
await Promise.all(promises);
```

**Reply by samimhamza-pgl** [reply]
ok thank you for your feedback, it was really helpful. we are looking forward to v1 release
